Libs/MRML/Core/vtkMRMLClipModelsNode.cxx

```diff
@@ -54,6 +55,9 @@ void vtkMRMLClipModelsNode::WriteXML(ostream& of, int nIndent)
   of << indent << " redSliceClipState=\"" << this->RedSliceClipState << "\"";
   of << indent << " yellowSliceClipState=\"" << this->YellowSliceClipState << "\"";
   of << indent << " greenSliceClipState=\"" << this->GreenSliceClipState << "\"";
+  of << indent << " clippingMethod=\"" << ((this->ClippingMethod == vtkMRMLClipModelsNode::
+    Straight) ? "Straight" : ((this->ClippingMethod == vtkMRMLClipModelsNode::
+    WholeCells) ? "WholeCells" : "WholeCellsWithBoundary")) << "\"";
```

Comment:
Please follow the pattern that used for example in vtkMRMLAbstractViewNode (https://github.com/Slicer/Slicer/blob/master/Libs/MRML/Core/vtkMRMLAbstractViewNode.cxx) to handle ruler type:
<pre>
//----------------------------------------------------------------------------
void vtkMRMLAbstractViewNode::WriteXML(ostream& of, int nIndent)
{
...    of << indent << " rulerType=\"" << this->GetRulerTypeAsString(this->RulerType) << "\"";
}

//----------------------------------------------------------------------------
void vtkMRMLAbstractViewNode::ReadXMLAttributes(const char** atts)
{
...
    else if (!strcmp(attName, "rulerType") && this->RulerEnabled)
      {
      int id = this->GetRulerTypeFromString(attValue);
      if (id<0)
        {
        vtkWarningMacro("Invalid rulerType: "<<(attValue?attValue:"(none)"));
        }
      else
        {
        this->RulerType = id;
        }
      }

//----------------------------------------------------------------------------
void vtkMRMLAbstractViewNode::PrintSelf(ostream& os, vtkIndent indent)
{ ...
  if (this->RulerEnabled)
    {
    os << indent << "Ruler type: " << this->GetRulerTypeAsString(this->RulerType) << "\n";
    }

//-----------------------------------------------------------
const char* vtkMRMLAbstractViewNode::GetRulerTypeAsString(int id)
{
  switch (id)
    {
    case RulerTypeNone: return "none";
    case RulerTypeThin: return "thin";
    case RulerTypeThick: return "thick";
    default:
      // invalid id
      return "";
    }
}

//-----------------------------------------------------------
int vtkMRMLAbstractViewNode::GetRulerTypeFromString(const char* name)
{
  if (name == NULL)
  {
    // invalid name
    return -1;
  }
  for (int i=0; i<RulerType_Last; i++)
    {
    if (strcmp(name, GetRulerTypeAsString(i))==0)
      {
      // found a matching name
      return i;
      }
    }
  // unknown name
  return -1;
}
</pre>